domain/src/main/java/com/example/domain/model/Result.kt

```diff
@@ -0,0 +1,16 @@
+package com.example.domain.model
+
+sealed class Result<out T : Any>
+data class Success<out T : Any>(val data: T) : Result<T>()
+data class Failure(val httpError: HttpError) : Result<Nothing>()
+
+class HttpError(val throwable: Throwable, val errorCode: Int = 0)
```

Comment:
Where is no network operation in this app

Reply:
Class modified as per need